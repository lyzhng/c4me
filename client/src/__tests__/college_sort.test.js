//configuration
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
configure({ adapter: new Adapter() });


import React from 'react';
import { mount, shallow, render } from 'enzyme';
import CollegeSearch from '../components/college/SearchColleges.jsx';
const fs = require('fs');
const colleges = JSON.parse(fs.readFileSync("./src/__tests__/colleges.json")).colleges;

describe('CollegeSearch', () => {
  it('should sort colleges by rank after clicking sort by rank', async() => {
    const component = mount((<CollegeSearch userid = "sam"/>));
    const sortCriteriaInput = component.find({ name: 'sortCriteria'});
    const sortButton = component.find('button').at(1);

  	//set component college state to colleges
  	component.state().colleges = colleges;
  	//set ascending to true
  	component.state().ascending = true;

  	//change sortCriteria to name and sort
  	sortCriteriaInput.simulate('change', {target : {name : "sortCriteria", value : "name"}});
  	sortButton.simulate('click');
  	for (let i = 0; i < component.state().colleges.length - 1; i ++)
  	{
  		let college1 = component.state().colleges[i];
  		let college2 = component.state().colleges[i + 1];

  		expect(college1.name.toLowerCase() <= college2.name.toLowerCase()).toEqual(true);	
  	}

  	//change sortCriteria to admission rate
  	sortCriteriaInput.simulate('change', {target : {name : "sortCriteria", value : "admissionRate"}});
  	sortButton.simulate('click');
  	for (let i = 0; i < component.state().colleges.length - 1; i ++)
  	{
  		let college1 = component.state().colleges[i];
  		let college2 = component.state().colleges[i + 1];

  		expect(college1.admission_rate <= college2.admission_rate).toEqual(true); 		
  	}

  	//change sortCriteria to cost of attendance
  	sortCriteriaInput.simulate('change', {target : {name : "sortCriteria", value : "costOfAttendance"}});
  	sortButton.simulate('click');
  	for (let i = 0; i < component.state().colleges.length - 1; i ++)
  	{
  		let college1 = component.state().colleges[i];
  		let college2 = component.state().colleges[i + 1];

  		expect(college1.cost.attendance.in_state <= college2.cost.attendance.in_state).toEqual(true);	
  	}

  	//change sortCriteria to ranking
  	sortCriteriaInput.simulate('change', {target : {name : "sortCriteria", value : "ranking"}});
  	sortButton.simulate('click');
  	for (let i = 0; i < component.state().colleges.length - 1; i ++)
  	{
  		let college1 = component.state().colleges[i];
  		let college2 = component.state().colleges[i + 1];

  		expect(college1.ranking <= college2.ranking).toEqual(true);
  	}
    
  });
});

