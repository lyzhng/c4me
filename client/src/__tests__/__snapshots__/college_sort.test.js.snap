// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollegeSearch should sort colleges by rank after clicking sort by rank 1`] = `
<SearchColleges
  userid="sam"
>
  <div
    className="container"
  >
    <h1
      className="text-center"
    >
      Search for Colleges!
    </h1>
    <div
      className="row"
    >
      <div
        className="col-4"
      >
        <div>
          strict:
          <input
            onClick={[Function]}
            type="checkbox"
          />
        </div>
        <div>
          rankingLower
          <input
            name="rankingLower"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          rankingUpper
          <input
            name="rankingUpper"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          admissionRateLower
          <input
            name="admissionRateLower"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          admissionRateUpper
          <input
            name="admissionRateUpper"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          sizeLower
          <input
            name="sizeLower"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          sizeUpper
          <input
            name="sizeUpper"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          satMathLower
          <input
            name="satMathLower"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          satMathUpper
          <input
            name="satMathUpper"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          satEngLower
          <input
            name="satEngLower"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          satEngUpper
          <input
            name="satEngUpper"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          actLower
          <input
            name="actLower"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          actUpper
          <input
            name="actUpper"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          major1
          <input
            name="major1"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div>
          major2
          <input
            name="major2"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div>
          costOfAttendance
          <input
            name="costOfAttendance"
            onChange={[Function]}
            type="number"
          />
        </div>
        <div>
          location
          <select
            name="location"
            onChange={[Function]}
          >
            <option
              value=""
            >
              No preference
            </option>
            <option
              value="northeast"
            >
              Northeast
            </option>
            <option
              value="midwest"
            >
              Midwest
            </option>
            <option
              value="south"
            >
              South
            </option>
            <option
              value="west"
            >
              West
            </option>
          </select>
        </div>
        <div>
          sort by
          <select
            name="sortCriteria"
            onChange={[Function]}
          >
            <option
              value=""
            >
              No option
            </option>
            <option
              value="name"
            >
              by name
            </option>
            <option
              value="admissionRate"
            >
              by admission rate
            </option>
            <option
              value="costOfAttendance"
            >
              by cost
            </option>
            <option
              value="ranking"
            >
              by ranking
            </option>
          </select>
          ascending:
          <input
            onClick={[Function]}
            type="checkbox"
          />
        </div>
        <div>
          <button
            onClick={[Function]}
          >
            Apply Filters
          </button>
          <button
            onClick={[Function]}
          >
            Apply Sort
          </button>
        </div>
      </div>
      <div
        className="col-8"
      >
        <label
          htmlFor="name"
        >
          Filter by Name:
        </label>
        <input
          name="name"
          onChange={[Function]}
          type="text"
        />
        <button
          onClick={[Function]}
        >
          Search
        </button>
      </div>
    </div>
  </div>
</SearchColleges>
`;
